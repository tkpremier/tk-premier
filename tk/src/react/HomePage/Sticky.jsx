import React, { createRef, useEffect } from 'react';
import PropTypes from 'prop-types';

const Sticky = props => {
  const ref = createRef(null);
  // useEffect(() => {
  //   $(ref.current).draggable({ containment: '.homepage' });
  // }, []);
  // const onClose = () => {
  //   $(ref.current).remove();
  // };
  const imgSrc = !props.source.includes('http')
    ? `https://www.phillips.com/Xigen/image.ashx?path=${decodeURIComponent(this.props.source)}`
    : props.source;
  return (
    <div className="sticky" ref={ref}>
      <div className="sticky__header">
        <button className="sticky__close" type="button" />
      </div>
      <a href={props.wrappingLink} title={props.altText}>
        <img className="sticky__img" src={imgSrc} alt={props.altText} title={props.altText} />
      </a>
    </div>
  );
};

Sticky.defaultProps = {
  altText: '',
  source: '',
  wrappingLink: ''
};

Sticky.propTypes = {
  altText: PropTypes.string,
  source: PropTypes.string,
  wrappingLink: PropTypes.string
};

export default Sticky;
