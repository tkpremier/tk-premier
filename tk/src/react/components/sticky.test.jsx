import React from 'react';
import Sticky from '../HomePage/Sticky';
import renderer from 'react-test-renderer';

it('should render a sticky', () => {
    const sticky = renderer.create(
        <stickyMarkup
            source="\\phillipsds\website\Media_preview\FeaturedPromo\FeaturedPromoImage_nextPhotosTCA.png"
            wrappingLink="https://live.phillips.com/"
            altText="Next Photos and Contemporary"
            imgSrc="/Xigen/image.ashx?path=\\phillipsds\website\Media_preview\FeaturedPromo\FeaturedPromoImage_nextPhotosTCA.png"
        />
    );
    const tree = sticky.toJSON();
    expect(tree).toMatchSnapshot();
});
