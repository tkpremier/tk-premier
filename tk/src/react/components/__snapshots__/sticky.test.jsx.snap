// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a sticky 1`] = `
<stickyMarkup
  altText="Next Photos and Contemporary"
  imgSrc="/Xigen/image.ashx?path=\\\\\\\\phillipsds\\\\website\\\\Media_preview\\\\FeaturedPromo\\\\FeaturedPromoImage_nextPhotosTCA.png"
  source="\\\\\\\\phillipsds\\\\website\\\\Media_preview\\\\FeaturedPromo\\\\FeaturedPromoImage_nextPhotosTCA.png"
  wrappingLink="https://live.phillips.com/"
/>
`;
